core/drivers/nvmem/nvmem_otp_key.c

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2023, Microchip
+ */
+
+#include <drivers/nvmem.h>
+#include <io.h>
+#include <kernel/dt.h>
+#include <kernel/huk_subkey.h>
+#include <kernel/tee_common_otp.h>
+#include <malloc.h>
+#include <matrix.h>
+#include <sama5d2.h>
+#include <tee_api_defines.h>
+#include <tee_api_types.h>
+#include <types_ext.h>
+#include <trace.h>
+
+static uint8_t *hw_unique_key;
+static uint8_t *die_id;
+static size_t die_id_len;
+
+TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
+{
+	if (!hw_unique_key)
+		return TEE_ERROR_NO_DATA;
+
+	memcpy(hwkey->data, hw_unique_key, HW_UNIQUE_KEY_LENGTH);
```

Comment:
Since the HUK is a quite sensible information, I think this function should read the NVMEM node straight instead of reading a copy of the HUK read and stored at probe time.